### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by ethanpinto**
Hi,

This is my initial PR to be merged into the topic recognition branch of the Pattern Analysis 2023 repository! It contains code for a Siamese Neural Network Classifier for the ADNI brain dataset. The following files have been included:
- dataset.py: loads the ADNI data into dataloaders for training and testing the model.
- train.py: trains the SNN and MLP, and saves the loss plots.
- modules.py: contains the architecture of the SNN and MLP.
- predict.py: contains code for inference and accuracy calculation.

Kind regards,
Ethan Pinto

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- some details provided on algorithm and training choices
- **loss and accuracy plots are shown only for the training set**, this is not very useful and doesn't say anything about the generalisation/actual performance of the model

Commit messages: Poor

Code: 
- good design
- sufficiently commented
- algorithm seem to be implemented correctly
- no hyperparameter tuning or attempts to improve performance

Functionality/Performance: 
- difficult to judge actual performance of the siamese and classifier without metrics/loss or accuracy plots on validation/test set
- classifier only achieves 60% of accuracy on test set

General comments:
- algorithm solves the problem apporpriately although performance is very poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem but no validation evidence -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no validation results -1
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, move files from parent directory into project directory, restore repo README -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
